### ISSUE TITLE
Give more respect to the whitelist param

### ISSUE THREAD

**Issue by rthallisey** [openshift/ansible-service-broker#685]
# Bug
The issue https://github.com/ansibleplaybookbundle/ansible-playbook-bundle/issues/215 pointed out that regardless of whitelist value, we always expect the suffix ```-apb``` in order for the apb to be seen by the ansible-service-broker. The relevant code is [here](https://github.com/openshift/ansible-service-broker/blob/master/pkg/registries/registry.go#L165).

I think it's worth to consider giving more power to the whitelist param.  I think it will be less error prone if the user had full control even though it's likely they will use the suffix ```-apb```. 

We can take on this issue as part of the working going into https://github.com/openshift/ansible-service-broker/issues/684

**Comment by eriknelson** [comment]
This dates all the way back to CAP actually; IIRC, when we were first investigating how to do discovery, there were some limitations with the dockerhub API where the process was to query a repo for a set of matching image names, then pull those images. Obviously we didn't want to pull all the images to check to see if they were an APB, so we made a decision and said "an atomicapp ends with -atomic" (this was when things were called atomicapps). That let us make sure we weren't doing throwaway work.

Now, obviously we've learned a lot. Particularly, there should be an API that let's us pull and inspect the metadata without downloading the image. If an APB label is on that image, we can trust that it's an APB and include it in the set of image names to be filtered by the white/black lists.

IMO, we should remove the function that hard enforced a "-apb" suffix, and update the adapters to inspect the metadata and only report ImageNames that contain the label. It may require some shifting around of the `Adapter` interface, but that's how I'd do it. Hopefully we can get away from imposing a name convention on users; that's kind of ugly.
